src/framing/FramedReader.cpp

```diff
@@ -66,35 +69,31 @@ size_t FramedReader::readFrameLength() const {
   return frameLength;
 }
 
-void FramedReader::onSubscribeImpl(
-    std::shared_ptr<Subscription> subscription) noexcept {
-  CHECK(!streamSubscription_);
-  streamSubscription_ = std::move(subscription);
-  frames_->onSubscribe(shared_from_this());
+void FramedReader::onSubscribe(
+    yarpl::Reference<Subscription> subscription) noexcept {
+  SubscriberBase::onSubscribe(subscription);
+  subscription->request(kMaxRequestN);
```

Comment:
This "request" in here makes the Http2StreamReader to fail.

```
void Http2StreamHandler::request(int64_t n) noexcept {
  // since proxygen doesn't support flow control, there is no point in allowing
  // more than n = 1. The underalying http stream can be transported over
  // as many chunks as necessary and we will always call onNext with the chain
  // of all chunks we received so far
  CHECK_LE(n, 1) << "can be called with n <= 1";
```
So I have removed this line and tested that. It seems still working fine without this line. I will verify with @lehecka .